hatch_build.py

```diff
@@ -7,7 +7,7 @@
 
 BASE_DEPS = [
     # psycopg2 dependency installed in custom hatch_build.py
-    "dbt-adapters>=0.1.0a1,<2.0",
+    "dbt-adapters @ git+https://github.com/dbt-labs/dbt-adapters.git@dbeatty/unknown-data-type-code-8912",
```

Comment:
We need to revert this before merging. However, this causes an issue. If we don't point to GH, the integration tests will fail. If we do point to GH, then we can't release because PyPI doesn't not allow pointing to mutable versions (i.e. GH branches/tags). We need to resolve this in general (beyond the scope of this PR). I think for now, I would update this to point to `main` before merging and we'll discuss how to handle this in general prior to releasing.

Reply:
We've since reverted the `hatch_build` approach, this should be back in `pyproject.toml` now.